Build errors for stdbuf on OpenBSD

On OpenBSD current/amd64 with Rust 1.90.0 + fix from PR #9158 (fix build/tests for `uptime` on OpenBSD), there are some errors when trying to build `stdbuf`:

```sh
$ rustc -vV
rustc 1.90.0 (1159e78c4 2025-09-14) (built from a source tarball)
binary: rustc
commit-hash: 1159e78c4747b02ef996e55082b704c09b970588
commit-date: 2025-09-14
host: x86_64-unknown-openbsd
release: 1.90.0
LLVM version: 19.1.7

$ cargo build -v --no-default-features --features unix
(...)
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `platform`
  --> src/uu/stdbuf/build.rs:62:47
   |
62 |         let lib_name = format!("libstdbuf{}", platform::DYLIB_EXT);
   |                                               ^^^^^^^^ use of unresolved module or unlinked crate `platform`
   |
   = help: if you wanted to use a crate named `platform`, use `cargo add platform` to add it to your `Cargo.toml`

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `platform`
   --> src/uu/stdbuf/build.rs:105:43
    |
105 |     let lib_name = format!("libstdbuf{}", platform::DYLIB_EXT);
    |                                           ^^^^^^^^ use of unresolved module or unlinked crate `platform`
    |
    = help: if you wanted to use a crate named `platform`, use `cargo add platform` to add it to your `Cargo.toml`

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `platform`
   --> src/uu/stdbuf/build.rs:106:69
    |
106 |     let dest_path = Path::new(&out_dir).join(format!("libstdbuf{}", platform::DYLIB_EXT));
    |                                                                     ^^^^^^^^ use of unresolved module or unlinked crate `platform`
    |
    = help: if you wanted to use a crate named `platform`, use `cargo add platform` to add it to your `Cargo.toml`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `uu_stdbuf` (build script) due to 3 previous errors
```

--- a/src/uu/stdbuf/build.rs
+++ b/src/uu/stdbuf/build.rs
@@ -14,6 +14,7 @@ use std::process::Command;
     target_os = "android",
     target_os = "freebsd",
     target_os = "netbsd",
+    target_os = "openbsd",
     target_os = "dragonfly"
 ))]
 mod platform {

--- a/src/uu/stdbuf/src/libstdbuf/src/libstdbuf.rs
+++ b/src/uu/stdbuf/src/libstdbuf/src/libstdbuf.rs
@@ -27,7 +27,15 @@ pub unsafe extern "C" fn __stdbuf_get_stdin() -> *mut FILE {
         unsafe { __stdinp() }
     }
 
-    #[cfg(not(any(target_os = "macos", target_os = "freebsd")))]
+    #[cfg(target_os = "openbsd")]
+    {
+        unsafe extern "C" {
+            static mut __stdin: *mut FILE;
+        }
+        unsafe { __stdin }
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "freebsd", target_os = "openbsd")))]
     {
         unsafe extern "C" {
             static mut stdin: *mut FILE;
@@ -48,7 +56,15 @@ pub unsafe extern "C" fn __stdbuf_get_stdout() -> *mut FILE {
         unsafe { __stdoutp() }
     }
 
-    #[cfg(not(any(target_os = "macos", target_os = "freebsd")))]
+    #[cfg(target_os = "openbsd")]
+    {
+        unsafe extern "C" {
+            static mut __stdout: *mut FILE;
+        }
+        unsafe { __stdout }
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "freebsd", target_os = "openbsd")))]
     {
         unsafe extern "C" {
             static mut stdout: *mut FILE;
@@ -69,7 +85,15 @@ pub unsafe extern "C" fn __stdbuf_get_stderr() -> *mut FILE {
         unsafe { __stderrp() }
     }
 
-    #[cfg(not(any(target_os = "macos", target_os = "freebsd")))]
+    #[cfg(target_os = "openbsd")]
+    {
+        unsafe extern "C" {
+            static mut __stderr: *mut FILE;
+        }
+        unsafe { __stderr }
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "freebsd", target_os = "openbsd")))]
     {
         unsafe extern "C" {
             static mut stderr: *mut FILE;

--- a/src/uu/stdbuf/src/stdbuf.rs
+++ b/src/uu/stdbuf/src/stdbuf.rs
@@ -35,6 +35,7 @@ mod options {
         target_os = "android",
         target_os = "freebsd",
         target_os = "netbsd",
+        target_os = "openbsd",
         target_os = "dragonfly"
     )
 ))]
@@ -82,6 +83,7 @@ enum ProgramOptionsError {
     target_os = "android",
     target_os = "freebsd",
     target_os = "netbsd",
+    target_os = "openbsd",
     target_os = "dragonfly"
 ))]
 fn preload_strings() -> UResult<(&'static str, &'static str)> {
@@ -98,6 +100,7 @@ fn preload_strings() -> UResult<(&'static str, &'static str)> {
     target_os = "android",
     target_os = "freebsd",
     target_os = "netbsd",
+    target_os = "openbsd",
     target_os = "dragonfly",
     target_vendor = "apple"
 )))]

